#! /usr/bin/env python

PACKAGE='planner'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name                             Type      Level Description                   Default  Min      Max
gen.add("target_surge",                  double_t, 0,    "Desired Vehicle Speed",      1,       0,       10)
gen.add("target_depth",                  double_t, 0,    "Desired Vehicle Depth",      0,       0,       100)
gen.add("target_roll",                   double_t, 0,    "Desired Vehicle Roll",       0,       -90,     90)
gen.add("target_pitch",                  double_t, 0,    "Desired Vehicle Pitch",      0,       -90,     90)
gen.add("target_yaw",                    double_t, 0,    "Desired Vehicle Yaw",        0,       0,       360)
gen.add("target_lat",                    double_t, 0,    "Desired Latitude",           0,      -90,      90)
gen.add("target_lon",                    double_t, 0,    "Desired Longitude",          0,      -180,     180)
gen.add("dynamic_reconfigure_enable",    bool_t, 0,      "Enable Dynamic Reconfigure", False)
exit(gen.generate(PACKAGE, "planner", "plannerParams"))
